### PR TITLE
Feature/mn 366 upgrade torch

### PR DESCRIPTION
**Pull request by ahirner**
# MN-366 Upgrade Torch

## How to test

On a GPU machine with `cuda >= 11.0`: 

```shell
docker pull moonvision/dev-moonbox:cuda-latest
docker run -it --rm --gpus all moonvision/dev-moonbox:cuda-latest python -c "\
import torch, torchvision;\
assert torch.cuda.is_available();\
print(torch.__version__, torchvision.__version__)
"
```
Expect: `1.11.0+cu113 0.12.0+cu113`.

### CODE REVIEW

File: docker/python-base/install_conda.sh
Comment (ahirner) [inline docker/python-base/install_conda.sh:4]: [Miniforge](https://github.com/conda-forge/miniforge) makes it harder to mess up conda channel preference.

[mamba](https://github.com/mamba-org/mamba) has crucial tooling for dependency fixes, like `mamba repoquery whoneeds`.

File: .github/workflows/build.yml
Comment (truh) [inline .github/workflows/build.yml]: Will you revert this?

File: docker/python-base/environment.yml
Comment (truh) [inline docker/python-base/environment.yml:12]: what does  `=*=*` mean? can you add a comment

Comment (ahirner) [reply docker/python-base/environment.yml:12]: done

File: .github/workflows/build.yml
Comment (ahirner) [reply .github/workflows/build.yml]: Yes, before merging downstream PRs. This requires approval of all related PRs.

**Review by truh — APPROVED** [review]

**Comment by truh** [comment]
Looks good in isolation. Might unapprove if I find a problem in the other PRs which might be traced down to changes here.

**Comment by ahirner** [comment]
Thx for the fixes in master. I'll perform the merges/changes across repos outside working hours later this week.

**Pull request merged**
